### PR TITLE
Support table comment for Snowflake connector

### PR DESCRIPTION
**Pull request by chenjian2664**
Allow create table with comment.
Also allow to set table comment for Snowflake tables.

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Snowflake
* Add support for creating tables with table comments.
* Add support for commenting on the existing tables. 
```

### CODE REVIEW

File: plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java [hunk 1]
```diff
@@ -319,9 +325,42 @@ public boolean isTopNGuaranteed(ConnectorSession session)
 
     @Override
     public Optional<String> getTableComment(ResultSet resultSet)
+            throws SQLException
     {
-        // Don't return a comment until the connector supports creating tables with comment
-        return Optional.empty();
+        return Optional.ofNullable(emptyToNull(resultSet.getString("REMARKS")));
```
Comment (ebyhr) [inline plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java:330]: Please leave comment why using `emptyToNull`.

File: plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java [hunk 2]
```diff
@@ -319,9 +325,42 @@ public boolean isTopNGuaranteed(ConnectorSession session)
...
+    @Override
+    protected List<String> createTableSqls(RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
+    {
+        checkArgument(tableMetadata.getProperties().isEmpty(), "Unsupported table properties: %s", tableMetadata.getProperties());
+        ImmutableList.Builder<String> createTableSqlsBuilder = ImmutableList.builder();
+        createTableSqlsBuilder.add(format("CREATE TABLE %s (%s)", quoted(remoteTableName), join(", ", columns)));
+        Optional<String> tableComment = tableMetadata.getComment();
+        if (tableComment.isPresent()) {
+            createTableSqlsBuilder.add(buildTableCommentSql(remoteTableName, tableComment));
+        }
```
Comment (ebyhr) [inline plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java:342]: Snowflake accepts a comment in CREATE TABLE statement as far as I know. No need to call COMMENT ON TABLE.

**Review by ebyhr — CHANGES_REQUESTED** [review]

**Comment by ebyhr** [comment]
/test-with-secrets sha=be5b0035e67d1d06242f3ac33cc3feb4cf53828b

**Review by ebyhr — APPROVED** [review]

**Comment by mosabua** [comment]
Amazing @chenjian2664 .. you are on a roll!

**Pull request merged**